src/pages/cookbooks/[cookbookId]/index.tsx

```diff
@@ -3,24 +3,39 @@ import { useRouter } from 'next/router';
 import { ROUTES } from '../../../constants/constants';
 import HttpService from '../../../utils/HttpService';
 import { useStore } from '../../../store/store';
-
-const init = async (cookbookId?: string | string[]) => {
-  if (cookbookId != null) {
-    const cookbook = await HttpService.getById(ROUTES.COOKBOOKS, cookbookId);
-    const guides = await HttpService.get(ROUTES.GUIDES(cookbook._id));
-    useStore.setState({ cookbook, guides });
-  }
-};
+import { useState, useEffect } from 'react';
+import { TwitchSidebar } from '../../../components/Sidebar/TwitchSidebar/TwitchSidebar';
 
 export const CookbookHome: React.FC = () => {
   const router = useRouter();
   const { cookbookId } = router.query;
+  const [cookbook, setCookbook] = useState(useStore.getState().cookbook);
+  const [streams, setstreams] = useState(useStore.getState().cookbook?.streams);
+
+  useEffect(() => {
+    const init = async (cookbookId?: string | string[]) => {
+      if (cookbookId != null) {
+        const cookbook = useStore
+          .getState()
+          .cookbooks.find(cookbook => cookbook._id === cookbookId);
+
+        if (cookbook) {
+          setCookbook(cookbook);
+          const guides = await HttpService.get(ROUTES.GUIDES(cookbook._id));
+          useStore.setState({ cookbook, guides });
+          setstreams(cookbook.streams);
```

Comment:
I would assume this to be `setStream` instead of ` setstream`

Reply:
was just poorly named function in the useState, but yeah i renamed it